File: Blackjack text based/backup/Blackjack.pas
program Blackjack;
uses crt;
label
  apostar, decision, escoge, inicio;
procedure check(var num:integer);
begin
    case num of
    11:
      begin
      writeln('J');
      num:=10;
      end;
    12:
      begin
      writeln('Q');
      num:=10;
      end;
    13:
      begin
      writeln('K');
      num:=10;
      end;
    1:
      begin
        writeln('A');
        num:=11;
      end
    else
      writeln(num);
    end;
end;
const
  diff:array[1..3] of string=('Facil','Medio', 'Dificil');
var
  pc, choice, aux,total,bankr,j,apuesta,dificultad,meta:integer;
  a,b:boolean;
begin
  randomize;
  inicio:
  clrscr;
  writeln('Bienvenido a Blackjack!');
  writeln('Hay 3 modos de dificultad, en todos empiezas con $50');
  writeln;
  writeln('1. Facil: Tu meta son $250');
  writeln;
  writeln('2. Medio: Tu meta son $500');
  writeln;
  writeln('3. Dificil: Tu meta son $1000');
  escoge:
  readln(dificultad);
  case dificultad of
  1: meta:=250;
  2: meta:=500;
  3: meta:=1000;
  else
    begin
      writeln('Esa opcion no existe');
      goto escoge;
    end;
  end;

  writeln('Escogiste el modo ', diff[dificultad]);
  writeln('Pierdes si te quedas sin dinero');
  writeln('Suerte!');
  readkey;
  bankr := 50;
  apostar:
  if (bankr <> 0) and (bankr < meta) then
  begin
    a:=false;
    b:= false;
    pc:=0;
    total:=0;
    j:=0;
    clrscr;
    writeln('Tu dinero:', bankr);
    writeln('Te faltan ', meta-bankr,' para ganar!');
    writeln; writeln;
    writeln('Cuanto apuestas?');
    readln(apuesta);

    if apuesta>bankr then
    begin
       writeln('No tienes tanto dinero');
       readkey;
       goto apostar;
    end;
    bankr:= bankr-apuesta;
    clrscr;
    writeln('Te reparten:');
    //aux := random(13)+1;
    aux := 1;
    if aux = 1 then a:= true;
    check(aux);
    total:=total+aux;
    //aux := random(13)+1;
    aux := 1;
    if aux = 1 then a:= true;
    check(aux);

    total:=total+aux;
    if (total > 21) AND (a = true) then
       total := total - 10;
    if total = 21 then
    begin
       writeln('Blackjack!');
       bankr:=bankr + (apuesta*2);
       goto apostar;
    end;

    decision:
    writeln;
    writeln('Tienes: ', total);
    writeln;
    writeln('Que deseas hacer?');
    writeln('1. Pedir');
    writeln('2. Quedarse');
    readln (choice);
    case choice of
    1:
      begin
        //aux:=random(13)+1;
        aux:=1;
        if aux = 1 then a:= true;
        writeln;
        writeln('Te reparten:');
        check(aux);
        total:=total+aux;
        if (total > 21) AND (a = true) then
        begin
             total := total - 10;
             a:=false;
        end;
        if total > 21 then
        begin
           writeln('Te pasaste!');
           readkey;
           goto apostar;
        end
        else goto decision;
      end;
    2:
      begin
      clrscr;
      writeln('Tu total es ', total);
      writeln;
      writeln;
      writeln('La pc tiene... ');
      writeln;
      writeln;
      while pc < 17 do
      begin
        j:=j+1;
        aux:=random(13)+1;
        if aux = 1 then b:= true;
        write('Carta ', j);
        write('   '); check(aux);
        readkey;
        pc := aux + pc;
        if (pc > 21) AND (b = true) then
        begin
             pc := pc- 10;
             b:=false;
        end;
      end;
      writeln;
      writeln('El total de la pc es ', pc);
      readkey;
      if pc > 21 then
      begin
         writeln;
         writeln('Tu ganas!');
         bankr:=bankr+apuesta*2;
         readkey;
         goto apostar;
      end
      else
          begin
              if pc>total then
                begin
                  writeln;
                  writeln('Tu pierdes!');
                  readkey;
                  goto apostar;
                end
              else if total>pc then
                begin
                  writeln;
                  writeln('Tu ganas!');
                  bankr := bankr+apuesta*2;
                  readkey;
                  goto apostar;
                end
              else
                begin
                  writeln;
                  writeln('Empate!');
                  bankr:=bankr+apuesta;
                  readkey;
                  goto apostar;
                end;
          end;
       end;
    end;

  end;
  clrscr;
  if bankr=0 then
  begin
    writeln('Te quedaste sin dinero!');
    writeln;
    writeln('Jugar de nuevo?');
    writeln('1. Si');
    writeln('2. No');
    readln(choice);
    case choice of
    1: goto inicio;
    end;
  end
  else
  begin
       writeln('Tu total: ',bankr);
       writeln('Felicidades! Ganaste el modo ',diff[dificultad],'!');
       writeln;
       writeln('Jugar de nuevo?');
       writeln('1. Si');
       writeln('2. No');
       readln(choice);
       case choice of
       1: goto inicio;
       end;
  end;


end.

File: Blackjack GUI (HTML-CSS-JS)/Blackjack.js
var bankr=50;
var total = 0;
var pc;
var aux;
var carta;
var valor;
var a;
var i;
var j;
var apuesta;
var meta = 250;
document.getElementById("bankr").innerHTML=bankr;
document.getElementById("jugador").style.visibility="hidden";

carta = document.getElementsByClassName("card");
valor = document.getElementsByClassName("number");
var cartapc = document.getElementsByClassName("card2");
var valor2 = document.getElementsByClassName("number2");

function dificultad(num) {
	if (num == 1) {
		meta = 250;
		alert("Seleccionado modo facil!");
	}
	else if (num == 2) {
		meta = 500;
		alert("Seleccionado modo medio!");
	}
	else {
		meta = 1000;
		alert("Seleccionado modo dificil!");
	}
}


function check(x) {
	switch (x) {
		case 11:
			x = 10;
			valor[i].innerHTML="J";
			break;
		case 12:
			x = 10;
			valor[i].innerHTML="Q";
			break;
		case 13:
			x = 10;
			valor[i].innerHTML="K";
			break;
		case 1:
			x = 11;
			valor[i].innerHTML="A";
			break;
		default:
			x=x;
			valor[i].innerHTML=x;
			break;
	}

	return x;
}

function checkpc(x) {
	switch (x) {
		case 11:
			x = 10;
			valor2[j].innerHTML="J";
			break;
		case 12:
			x = 10;
			valor2[j].innerHTML="Q";
			break;
		case 13:
			x = 10;
			valor2[j].innerHTML="K";
			break;
		case 1:
			x = 11;
			valor2[j].innerHTML="A";
			break;
		default:
			x=x;
			valor2[j].innerHTML=x;
			break;
	}

	return x;
}


function jugar() {

		for (i = 0; i<carta.length; i++) {
			carta[i].style.visibility="hidden";
			cartapc[i].style.visibility="hidden";
		}
		document.getElementById("totalpc").style.visibility="hidden";
		document.getElementById("bienvenida").style.visibility="hidden";
		document.getElementById("dinero").style.visibility="visible";
		document.getElementById("meta").innerHTML=meta;
		if (bankr != 0 && bankr < meta) {
			document.getElementById("jugador").style.visibility="hidden";
			i = 0;
			a=false;
			b=false;
			j=0;
			pc=0;
			total=0;
			apuesta = prompt("Cuanto quieres apostar? Tienes " + bankr);
			if (apuesta > bankr) {
				alert("No tienes tanto dinero");
				jugar();
			}

			else {
				bankr-=apuesta;
				document.getElementById("jugador").style.visibility="visible";
				document.getElementById("bankr").innerHTML=bankr;
				aux = Math.floor(Math.random(13)*13)+1;
				if (aux == 1) {a=true;}
				aux = check(aux);
				carta[i].style.visibility="visible";
				total += aux;

				i++;

				aux = Math.floor(Math.random(13)*13)+1;
				if (aux == 1) {a=true;}
				aux = check(aux);
				carta[i].style.visibility="visible";
				total += aux;
				if (total > 21 && a==true) {total -= 10;} 
				document.getElementById("total").innerHTML=total;
				i++;
			}
		}
		else if (bankr == 0) {
			alert("Te quedaste sin dinero! Refresca la pagina si quieres volver a jugar");
		}
		else if (bankr >= meta) {
			alert("Ganaste!");
		}
	}

function pedir() {
	aux = Math.floor(Math.random(13)*13)+1;
	if (aux == 1) {a=true;}
	aux = check(aux);
	total += aux;
	document.getElementById("total").innerHTML=total;
	carta[i].style.visibility="visible";
	i++;
	if (total > 21 && a==true) {
		total -= 10; 
		a=false;
		document.getElementById("total").innerHTML=total;
	} 
	if (total>21) {
		alert("Sacaste "+ aux +", tienes " + total + ", te pasaste!");
		jugar();
	}
}

function quedarse() {
	document.getElementById("totalpc").style.visibility="visible";
	while (pc < 17) {
		aux = Math.floor(Math.random(13)*13)+1;
		if (aux == 1) {b=true;}
		aux=checkpc(aux);
		pc+=aux;
		cartapc[j].style.visibility="visible";
		j++;
		if (pc > 21 && b==true) {pc -= 10; b=false;} 

	}
	document.getElementById("totalpc").innerHTML=pc;
	if (pc>21) {
		document.getElementById("totalpc").innerHTML=pc;
		setTimeout(function(){alert("Tu ganas!");bankr+=apuesta*2;document.getElementById("bankr").innerHTML=bankr;jugar();},1000);
	}
	else {
		document.getElementById("totalpc").style.visibility="visible";
		document.getElementById("totalpc").innerHTML=pc;
		comparar(pc,total);
	}
}

function comparar(pc,total) {
	if (pc>total) {
		setTimeout(function(){alert("Tu pierdes!");jugar();},1000);
	}
	else if (total > pc) {
		setTimeout(function(){alert("Tu ganas!");bankr+=apuesta*2;document.getElementById("bankr").innerHTML=bankr;jugar();},1000);

	}
	else if (total == pc) {
		setTimeout(function(){alert("Empate!");bankr+=apuesta*1;document.getElementById("bankr").innerHTML=bankr;jugar();},1000);
	}
}
